type/__clean_path/explorer/list

```diff
@@ -25,16 +24,82 @@ else
     path="/$__object_id"
 fi
 
-[ ! -d "$path" ] && exit 0
+if [ ! -d "$path" ]
+then
+    exit 0
+fi
 
-pattern="$( cat "$__object/parameter/pattern" )"
+set -- 'find' "$path" '-mindepth' 1
 
-if [ -f "$__object/parameter/exclude" ]
+# deprecated
+if [ -f "$__object/parameter/pattern" ] || [ -f "$__object/parameter/exclude" ]
 then
-    exclude="$( cat "$__object/parameter/exclude" )"
+    set -- "$@" '-maxdepth' 1
 
-    find "$path" -mindepth 1 -maxdepth 1 -regex "$pattern" \
-        -and -not -regex "$exclude"
-else
-    find "$path" -mindepth 1 -maxdepth 1 -regex "$pattern"
+    if [ -f "$__object/parameter/pattern" ]
+    then
+        cp "$__object/parameter/pattern" "$__object/parameter/rm-regex"
+    fi
+
+    if [ -f "$__object/parameter/exclude" ]
+    then
+        cp "$__object/parameter/exclude" "$__object/parameter/not-regex"
+    fi
+fi
+# /deprecated
+
+if [ -f "$__object/parameter/rm-name" ]
+then
+    while read -r rm_name
+    do
+        set -- "$@" '-name' "$rm_name"
+    done \
+        < "$__object/parameter/rm-name"
+fi
+
+if [ -f "$__object/parameter/rm-path" ]
+then
+    while read -r rm_path
+    do
+        set -- "$@" '-path' "$rm_path"
+    done \
+        < "$__object/parameter/rm-path"
+fi
+
+if [ -f "$__object/parameter/rm-regex" ]
+then
+    while read -r rm_regex
+    do
+        set -- "$@" '-regex' "$rm_regex"
+    done \
+        < "$__object/parameter/rm-regex"
+fi
+
+if [ -f "$__object/parameter/not-name" ]
+then
+    while read -r not_name
+    do
+        set -- "$@" '-not' '-name' "$not_name"
```

Comment:
Wouldn't it be possible to use `!` instead of `-not`?

`!` is POSIX, `-not` is not.

Reply:
Yes!